lang/python/wiredtiger/init.py

```diff
@@ -26,12 +26,17 @@
 # ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 # OTHER DEALINGS IN THE SOFTWARE.
 #
-
 # init.py
 #      This is installed as __init__.py, and imports the file created by SWIG.
 # This is needed because SWIG's import helper code created by certain SWIG
 # versions may be broken, see: https://github.com/swig/swig/issues/769 .
 # Importing indirectly seems to avoid these issues.
+
+# Restart this instance of python so it uses the most recent os.environ values
+def restart_python():
+    python = sys.executable
+    os.execl(python, python, *sys.argv)
```

Comment:
Should the 2nd arg of `os.execl()` be set to the running .py file (instead of the Python executable)?

Reply:
This is using the same process we do for ASan [here](https://github.com/wiredtiger/wiredtiger/blob/fe9a67d9aae542dee25090de550ff704e0200c01/test/suite/run.py#L533), but I'll try a patch with the file name as well.

Note we've intentionally split the TSan implementation out from the ASan implementation due to the `./wt`/`LD_PRELOAD` issues mentioned in [this thread](https://github.com/wiredtiger/wiredtiger/pull/10839#discussion_r1669705705).

Reply:
The patch seems to work, but there's still some tests in flight. I've updated to use the filepath instead in [4e55620](https://github.com/wiredtiger/wiredtiger/pull/10839/commits/4e556202492ca9611605025754bb8624aeee791e) and will roll it back if test failures start to appear.